kgforge/core/commons/sparql_query_builder.py

```diff
@@ -345,20 +367,111 @@ def _replace_in_sparql(
         return qr
 
     @staticmethod
-    def apply_limit_and_offset_to_query(query, limit, default_limit, offset, default_offset):
+    def apply_limit_and_offset_to_query(
+        query, limit, default_limit, offset, default_offset
+    ):
         if limit:
             query = SPARQLQueryBuilder._replace_in_sparql(
-                query, "LIMIT", limit, default_limit,
-                re.compile(r" LIMIT \d+", flags=re.IGNORECASE)
+                query,
+                "LIMIT",
+                limit,
+                default_limit,
+                re.compile(r" LIMIT \d+", flags=re.IGNORECASE),
             )
         if offset:
             query = SPARQLQueryBuilder._replace_in_sparql(
-                query, "OFFSET", offset, default_offset,
-                re.compile(r" OFFSET \d+", flags=re.IGNORECASE)
+                query,
+                "OFFSET",
+                offset,
+                default_offset,
+                re.compile(r" OFFSET \d+", flags=re.IGNORECASE),
             )
 
         return query
 
+    @staticmethod
+    def create_select_query(
+        vars_: List[str],
+        statements: Union[str, List[str]],
+        distinct: bool,
+        search_in_graph: bool,
+        union: bool = False,
+    ):
+        if not union:
+            all_statements = statements
+
+        elif not isinstance(statements, list):
+            raise AttributeError(
+                f"Expected statements argument to be of type  List[str] when union is True. Each item will go in a UNION clause: {statements}"
+            )
+        else:
+            all_statements = "}} UNION {{".join(statements)
+            all_statements = "{{" + all_statements + "}}"
+
+        where_clauses = (
+            f"{{ Graph ?g {{{all_statements}}}}}"
+            if search_in_graph
+            else f"{{{all_statements}}}"
+        )
+
+        join_vars_ = " ".join(vars_)
+        select_vars = f"DISTINCT {join_vars_}" if distinct else f"{join_vars_}"
+        return f"SELECT {select_vars} WHERE {where_clauses}"
+
 
 def _box_value_as_full_iri(value):
     return f"<{value}>" if is_valid_url(value) else value
+
+
+def build_shacl_query(
+    statements: List[str] = None,
+    defining_property_uri: str = None,
+    deprecated_property_uri: str = None,
```

Comment:
What's the point of including the deprecated statement if you don't add it to vars_?

Reply:
Deprecated should be false if deprecated_property_uri  is provided. No need for variable